### PR TITLE
Catalog movie view

### PR DESCRIPTION
**Pull request by jenna1k**
catalog-movie-view with mockMovies & movie.model

### CODE REVIEW

File: src/app/display/main/main.component.html [hunk 1]
```diff
@@ -43,97 +45,113 @@ <h3 class="mat-title">Principal Informations</h3>
...
         <article fxLayout="row" fxLayoutAlign="space-between start">
           <!-- First Column -->
           <ul>
+            <li><b>ISAN Number</b> {{ movie.main.isan }}</li>
+            <li><b>Length</b> {{ movie.main.length }} min</li>
+            <li><b>Color / Black & White</b> {{ movie.salesInfo.color }}</li>
             <li>
-              <b>ISAN Number</b> {{ main.isan }}
+              <b>European Qualification</b>
+              {{ movie.salesInfo.europeanQualification }}
```
Comment (GrandSchtroumpf) [inline src/app/display/main/main.component.html]: Keep in one line even if prettier is unhappy ;)

File: src/app/display/main/main.component.html [hunk 2]
```diff
@@ -43,97 +45,113 @@ <h3 class="mat-title">Principal Informations</h3>
...
             </li>
+          </ul>
+          <!-- Second Column -->
+          <ul>
             <li>
-              <b>Length</b> {{ main.length }} min
+              <b>Certifications</b>
+              <ng-container
```
Comment (GrandSchtroumpf) [inline src/app/display/main/main.component.html]: Keep it on one line.

File: src/app/display/main/main.component.html [hunk 3]
```diff
@@ -43,97 +45,113 @@ <h3 class="mat-title">Principal Informations</h3>
...
+                *ngFor="let certification of movie.salesInfo.certifications"
+              >
+                {{ certification }}
+              </ng-container>
             </li>
+            <li><b>Rating</b> {{ movie.salesInfo.pegi }}</li>
             <li>
-              <b>Color / Black & White</b> {{ main.colorBlackWhite }}
+              <b>International Premiere</b>
+              {{ movie.salesInfo.internationalPremiere.name }}
```
Comment (GrandSchtroumpf) [inline src/app/display/main/main.component.html]: Create a getter function in your component for that : 
```typescript
get internationalPremiere() {
  const {name, year, prize } = this.movie.salesInfo.internationalPremiere;
  return `${name} ${year} ${prize}`
}
```
And in your HTML : 
```html
<b>International Premiere</b> {{ internationalPremiere }}
```

File: src/app/display/main/main.component.html [hunk 4]
```diff
@@ -43,97 +45,113 @@ <h3 class="mat-title">Principal Informations</h3>
...
-              </li>
-            </ul>
         </article>
         <mat-divider></mat-divider>
         <!-- Festival Prizes -->
         <h3 class="mat-title">Festival Prizes</h3>
         <article fxLayout="row" fxLayoutAlign="space-between start">
-          <ng-container *ngFor="let festivalPrize of main.festivalPrizes">
-            {{ festivalPrize }}
+          <ng-container
```
Comment (GrandSchtroumpf) [inline src/app/display/main/main.component.html]: One line

File: src/app/display/main/main.component.html [hunk 5]
```diff
@@ -43,97 +45,113 @@ <h3 class="mat-title">Principal Informations</h3>
...
+            *ngFor="let festivalPrize of movie.festivalPrizes.prizes"
+          >
+            <p>
```
Comment (GrandSchtroumpf) [inline src/app/display/main/main.component.html]: Do : 
```html
<ng-container *ngIf="movie.festivalPrizes as festivalPrizes">
  <h3 class="mat-title">Festival Prizes</h3>
  <ng-container *ngFor="let prize of festivalPrizes.prizes">
    <p>{{ prize.name }}, {{ prize.year }}, {{ prize.prize }}</p>
  </ng-container>
</ng-container>
```

File: src/app/display/main/main.component.html [hunk 6]
```diff
@@ -43,97 +45,113 @@ <h3 class="mat-title">Principal Informations</h3>
...
+              {{ festivalPrize.prize }}
+            </p>
           </ng-container>
         </article>
         <mat-divider></mat-divider>
         <!-- Key Assets -->
         <h3 class="mat-title">Key Assets</h3>
-        <article *ngFor="let keyAsset of main.keyAssets">
-            <p>{{ keyAsset }}</p>
+        <article
```
Comment (GrandSchtroumpf) [inline src/app/display/main/main.component.html]: one line

File: src/app/display/main/main.component.html [hunk 7]
```diff
@@ -142,20 +160,20 @@ <h3 class="mat-title">Keywords</h3>
...
           <ul>
-            <li>
-              <b>Original language(s)</b> {{ main.originalLanguage }}
-            </li>
+            <li><b>Original language(s)</b> {{ movie.main.languages }}</li>
             <li>
               <b>Available dubbing(s)</b>
-              <ng-container *ngFor="let dub of main.dubs">
+              <ng-container *ngFor="let dub of movie.versionInfo.dubbings">
                 {{ dub }}
```
Comment (GrandSchtroumpf) [inline src/app/display/main/main.component.html]: `dubbing` instead of `dub`

File: src/app/display/main/main.component.html [hunk 8]
```diff
@@ -142,20 +160,20 @@ <h3 class="mat-title">Keywords</h3>
...
               </ng-container>
             </li>
           </ul>
           <ul>
             <li>
               <b>Available subtitle(s)</b>
-              <ng-container *ngFor="let language of main.languages">
+              <ng-container
+                *ngFor="let language of movie.versionInfo.subtitles"
```
Comment (GrandSchtroumpf) [inline src/app/display/main/main.component.html]: `subtitle` instead of `language`

File: src/app/movie.model.ts [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+  international?: string;
+}
+
+export interface Credit {
+  firstName: string;
+  lastName?: string;
+  creditRole?: string;
+}
+
+// endOfRights: number;
```
Comment (GrandSchtroumpf) [inline src/app/movie.model.ts]: Remove comments if you don't need them.

File: src/app/movie.model.ts [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+export interface MovieSalesAgentDeal {
+  rightsEnd: Date; //
+  territories: string[];
+  medias: string[];
+}
+
+export interface Movie {
+  // @todo #643 add new fields to Draw.io
```
Comment (GrandSchtroumpf) [inline src/app/movie.model.ts]: remove all comments.

File: src/app/pages/movie-view/movie-view.component.html [hunk 11]
```diff
@@ -1 +1 @@
-<catalog-display-main [main]="movie.main"></catalog-display-main>
\ No newline at end of file
+<catalog-display-main [movie]="movie"></catalog-display-main>
```
Comment (GrandSchtroumpf) [inline src/app/pages/movie-view/movie-view.component.html]: For now you can get ride of the sub component and do everything in the CatalogMovieViewComponent

File: src/app/display/main/main.component.html (hunk 4 above)
Comment (jenna1k) [reply src/app/display/main/main.component.html]: prettier refuse to make it one line 🤓

Comment (GrandSchtroumpf) [reply src/app/display/main/main.component.html]: Don't let prettier have the last word! You're smarter than a piece of software. Disable it on html files in the settings of VScode (ask Max how to do that).

Comment (jenna1k) [reply src/app/display/main/main.component.html]:
```json
// Set the default
"editor.formatOnSave": false,
// Enable per-language
"[javascript]": {
    "editor.formatOnSave": true
},
"[html]": {
    "editor.formatOnSave": true
}
```

like this?

Comment (GrandSchtroumpf) [reply src/app/display/main/main.component.html]: Hmm I would have done the opposite...

File: src/app/display/main/main.component.html [hunk 12]
```diff
@@ -43,97 +40,87 @@ <h3 class="mat-title">Principal Informations</h3>
...
-            <p>{{ keyAsset }}</p>
+        <article *ngFor="let keyAsset of movie.promotionalDescription.keyAssets">
+          <p>{{ keyAsset }}</p>
         </article>
         <mat-divider></mat-divider>
         <!-- Keywords -->
         <h3 class="mat-title">Keywords</h3>
         <article fxLayout="row">
-          <mat-chip-list aria-label="keywords" *ngFor="let keyword of main.keywords">
+          <mat-chip-list aria-label="keywords" *ngFor="let keyword of movie.promotionalDescription.keywords">
```
Comment (GrandSchtroumpf) [inline src/app/display/main/main.component.html]: *ngFor should be on the mat-chip

**Pull request merged**